src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -446,73 +354,49 @@ public async Task RestartAsync()
+        private async Task<ServiceBusProcessor> CreateMessageProcessorAsync(AzureServiceBusMessagePumpSettings messagePumpSettings)
         {
-            var rawConnectionString = await messagePumpSettings.GetConnectionStringAsync();
-            var serviceBusConnectionStringBuilder = new ServiceBusConnectionStringBuilder(rawConnectionString);
+            string rawConnectionString = await messagePumpSettings.GetConnectionStringAsync();
+            ServiceBusConnectionStringProperties serviceBusConnectionString = ServiceBusConnectionStringProperties.Parse(rawConnectionString);
 
-            MessageReceiver messageReceiver;
-            if (string.IsNullOrWhiteSpace(serviceBusConnectionStringBuilder.EntityPath))
+            var client = new ServiceBusClient(rawConnectionString);
             {
-                // Connection string doesn't include the entity so we're using the message pump settings
-                if (string.IsNullOrWhiteSpace(messagePumpSettings.EntityName))
+                ServiceBusProcessor processor;
+                if (string.IsNullOrWhiteSpace(serviceBusConnectionString.EntityPath))
                 {
-                    throw new ArgumentException("No entity name was specified while the connection string is scoped to the namespace");
-                }
-
-                messageReceiver = CreateReceiver(serviceBusConnectionStringBuilder, messagePumpSettings.EntityName, SubscriptionName);
-            }
-            else
-            {
-                // Connection string includes the entity so we're using that instead of the message pump settings
-                messageReceiver = CreateReceiver(serviceBusConnectionStringBuilder, serviceBusConnectionStringBuilder.EntityPath, SubscriptionName);
-            }
-
-            Namespace = messageReceiver.ServiceBusConnection?.Endpoint?.Host;
+                    // Connection string doesn't include the entity so we're using the message pump settings
+                    if (string.IsNullOrWhiteSpace(messagePumpSettings.EntityName))
+                    {
+                        throw new ArgumentException("No entity name was specified while the connection string is scoped to the namespace");
+                    }
 
-            ConfigurePlugins();
+                    processor = CreateProcessor(client, messagePumpSettings.EntityName, SubscriptionName);
+                }
+                else
+                {
+                    // Connection string includes the entity so we're using that instead of the message pump settings
+                    processor = CreateProcessor(client, serviceBusConnectionString.EntityPath, SubscriptionName);
+                }
 
-            RegisterClientInformation(messageReceiver.ClientId, messageReceiver.Path);
+                Namespace = serviceBusConnectionString.Endpoint?.Host;
 
-            return messageReceiver;
-        }
+                // TODO: show that we don't support plugins just yet
```

Comment:
Think we can remove this since it's documented, no? Or do you want to keep it as a placeholder?
If so, let's add an issue and link to it - That's fine for me.

Reply:
Yes, is documented; yes, I would want this as a placeholder for future work, and added an issue for this. Thanks! Resolving.